### PR TITLE
Update contracts.rst

### PR DESCRIPTION
**Pull request by mocamircea**
Extended example according to the discussed stuff, clarification & typos

### CODE REVIEW

File: docs/contracts.rst [hunk 1]
```diff
@@ -194,9 +194,16 @@ return parameter list for functions.
         function setData(uint a) internal { data = a; }
         uint public data;
     }
+    contract Caller {
+        function readData(){
+            C c = new C();
+            uint local = c.data(); // call accessor
```
Comment (chriseth) [inline docs/contracts.rst]: Accessors have not been introduced at this point.

Comment (mocamircea) [inline docs/contracts.rst]: yes but lines 198-200 rely on c.data() call which is related to accessor

Comment (mocamircea) [inline docs/contracts.rst]: Solution: a link to the Accessor Functions section below OR completely remove the comment?

Comment (chriseth) [inline docs/contracts.rst]: It is probably better to remove any mention of accessor functions here (instead use regular functions) and then explain the modifiers that are relevant for accessor functions in the section about accessor functions.

File: docs/contracts.rst [hunk 2]
```diff
@@ -194,9 +194,16 @@ return parameter list for functions.
         function setData(uint a) internal { data = a; }
         uint public data;
```
Comment (chriseth) [inline docs/contracts.rst:195]: As discussed, please remove the `public` from the variable here and change to:

```
   function setData() internal { _data = a; }
   function getData() { return _data; }
   uint _data;
```

**Pull request merged**